.github/workflows/deploy-staging.yml

```diff
@@ -0,0 +1,96 @@
+        with:
+          username: ${{secrets.DOCKERHUB_USERNAME}}
+          password: ${{secrets.DOCKERHUB_TOKEN}}
+          path: provisioning/base
+          repository: maelstromeous/applications
+          tags: ps2alerts-webocket-base
+
+      - name: Build staging image
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{secrets.DOCKERHUB_USERNAME }}
+          password: ${{secrets.DOCKERHUB_TOKEN}}
+          repository: maelstromeous/applications
+          dockerfile: provisioning/staging/Dockerfile
+          always_pull: true
+          tags: ps2alerts-websocket-staging-${{github.sha}}
+
+  deploy-terraform:
+    needs:
+      - build-docker-images
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Install the latest version of Terraform CLI and configure the Terraform
+      # CLI configuration file with a Terraform Cloud user API token
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          cli_config_credentials_token: ${{secrets.TERRAFORM_CLOUD_TOKEN}}
+
+      # Initialize a new or existing Terraform working directory by creating initial
+      # files, loading any remote state, downloading modules, etc.
+      - name: Terraform Init
+        run: terraform init
+        env:
+          AWS_ACCESS_KEY_ID: ${{secrets.AWS_ACCESS_KEY_ID}}
+          AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY}}
+
+      # Checks that all Terraform configuration files adhere to a canonical format
+      - name: Terraform Format
+        run: terraform fmt -check
+        env:
+          AWS_ACCESS_KEY_ID: ${{secrets.AWS_ACCESS_KEY_ID}}
+          AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY}}
+
+      # Generates an execution plan for Terraform, with the injected SHA
+      - name: Terraform Plan
+        run: terraform plan -var=' -var='checksum_version=${{github.sha}}'
+        env:
+          AWS_ACCESS_KEY_ID: ${{secrets.AWS_ACCESS_KEY_ID}}
+          AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY}}
+
+      # Apply the terraform plan and inject the SHA
+      # - name: Terraform Apply
+      # if: github.ref == 'refs/heads/staging' && github.event_name == 'push'
+      # run: terraform apply -auto-approve -var='checksum_version=${{ github.sha }}
+
+  # Unfortunately we need to do this step in order to provision the LE cert manager.
+  # Terraform currently doesn't support it :'(
+  deploy-cert-manager:
+    needs:
+      - deploy-terraform
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Install doctl
+        uses: digitalocean/action-doctl@v2
+        with:
+          token: ${{ secrets.DIGITAL_OCEAN_TOKEN }}
+
+      - name: Save DO kubeconfig
+        run: doctl kubernetes cluster kubeconfig save my-cluster
+
+      - name: Apply staging K8s Deployment
```

Comment:
Incorrect command - format below expects just a spec, file is actually a playbook

Reply:
Bollocks... need a new action for this :(